engine/runtime/src/main/java/org/enso/interpreter/node/callable/function/BlockNode.java

```diff
@@ -55,8 +57,15 @@ public static BlockNode buildSilent(ExpressionNode[] expressions, ExpressionNode
   @Override
   @ExplodeLoop
   public Object executeGeneric(VirtualFrame frame) {
+    var ctx = EnsoContext.get(this);
+    var nothing = ctx.getBuiltins().nothing();
     for (ExpressionNode statement : statements) {
-      statement.executeGeneric(frame);
+      var result = statement.executeGeneric(frame);
+      if (result != nothing) {
```

Comment:
Why not get rid of this if statement and use simply just `if (result instanceof DataflowError err)`. If it is an instance of `DataflowError`, it cannot be nothing.

Reply:
Indeed this seems redundant, can we remove it, or is there some reason that we don't see here?

Reply:
The reason is the _other part_ of #5430:
- what should happen when the value is not `Nothing`
- and it is not `DataflowError` (which is immediately returned)
- #5430 talks about such values, but
- I don't know what to do with them (they are not as simple as `DataflowError`)

In any case, that's for another PR. We don't need to solve it in this PR. Just my 2 Kč explanation of the `!= Nothing` check.